python/eos/analysis_file_description.py

```diff
@@ -0,0 +1,309 @@
+from dataclasses import dataclass, field
+import yaml as _yaml
+import copy as _copy
+import eos
+
+class Deserializable:
+    def __init__(self):
+        pass
+
+    @classmethod
+    def from_yaml(cls, yaml_data:str):
+        kwargs = _yaml.safe_load(yaml_data)
+        return cls.from_dict(**kwargs)
+
+    @staticmethod
+    def make(cls, **kwargs):
+        if not isinstance(cls, type):
+            raise ValueError(f'Argument cls=\'{cls}\' is not a type')
+
+        if not issubclass(cls, Deserializable):
+            raise ValueError(f'Class {cls.__name__} is not a subclass of Deserializable')
+
+        try:
+            result = cls(**kwargs)
+        except Exception as e:
+            raise ValueError(f'When creating {cls.__name__} from {kwargs}: {e}')
+
+        return result
+
+    @classmethod
+    def from_dict(cls, **kwargs):
+        return Deserializable.make(cls, **kwargs)
+
+
+
+class PriorDescription:
+    @staticmethod
+    def from_dict(**kwargs):
+        if 'constraint' in kwargs:
+            return Deserializable.make(ConstraintPriorDescription, **kwargs)
+        elif 'parameter' in kwargs and 'type' in kwargs:
+            _kwargs = _copy.deepcopy(kwargs)
+            _kwargs.pop('type')
+            if kwargs['type'] in ("uniform", "flat"):
+                return Deserializable.make(UniformPriorDescription, **_kwargs)
+            elif kwargs['type'] in ("scale",):
+                return Deserializable.make(ScalePriorDescription, **_kwargs)
+            elif kwargs['type'] in ("gauss", "gaussian"):
+                if "min" in kwargs:
+                    return Deserializable.make(CurtailedGaussianDescription, **_kwargs)
+                return Deserializable.make(GaussianPriorDescription, **_kwargs)
+            elif kwargs['type'] in ("poisson",):
+                return Deserializable.make(PoissonPriorDescription, **_kwargs)
+
+
+        raise ValueError('Unknown type of prior description')
+
+@dataclass
+class PoissonPriorDescription(Deserializable):
+    parameter:str
+    k:float
+    type: str = field(repr=False, init=False, default="poisson")
```

Comment:
Coding styles of lines 60+61 and 62 differ. I think elsewhere (in ``python/eos/tasks.py``) we use something like
```py
type:str=field(repr=False, init=False, default="poisson")
```